editor/src/clj/internal/util.clj

```diff
@@ -355,13 +355,19 @@
     (first coll)))
 
 (defn only-or-throw
-  "Returns the only element in coll, or throw an error if there are more than one element"
+  "Returns the only element in coll, or throw an error if the collection does not have exactly one element."
   [coll]
-  (or (let [first-element (first coll)]
-        (when (nil? (next coll))
-          first-element))
-      (throw (ex-info "The collection has more than one element."
-                      {:coll coll}))))
+  (cond
+    (next coll)
```

Comment:
Minor: `next` creates a seq from the coll, and `seq` below also creates a seq from the coll. Would be nice to reuse it. There is also `bounded-count` that can be used to count the first 2 elements without necessarily creating a seq at all if the coll is counted. If it's used only in tests, these micro-optimizations probably don't matter.

Reply:
I'll change it to use `bounded-count` since it is part of the `internal.util` module which might be used outside of tests as well.